src/pybamm/plotting/quick_plot.py

```diff
@@ -542,128 +545,104 @@ def plot(self, t, dynamic=False):
-                            color=self.colors[i],
-                            linestyle=linestyle,
-                            zorder=10,
-                        )
-                        variable_handles.append(self.plots[key][0][j])
-                    solution_handles.append(self.plots[key][i][0])
-                # add lines for boundaries between subdomains
-                for boundary in variable_lists[0][0].internal_boundaries:
-                    boundary_scaled = boundary * self.spatial_factor
-                    ax.axvline(boundary_scaled, color="0.5", lw=1, zorder=0)
+                ax.set_xlabel(f"{spatial_var_name} [{self.spatial_unit}]")
+
+                for idx, t_single in enumerate(t):
+                    t_in_seconds = t_single * self.time_scaling_factor
+
+                    for i, variable_list in enumerate(variable_lists):
+                        for j, variable in enumerate(variable_list):
+                            linestyle = (
+                                self.linestyles[i]
+                                if len(variable_list) == 1
+                                else self.linestyles[j]
+                            )
+                            (self.plots[key][i][j],) = ax.plot(
+                                self.first_spatial_variable[key],
+                                variable(t_in_seconds, **spatial_vars),
+                                color=time_colors[idx],  # Different color for each time
+                                linestyle=linestyle,
+                                label=f"t = {t_single:.2f} {self.time_unit}",  # Add time label
+                                zorder=10,
+                            )
+                            variable_handles.append(self.plots[key][0][j])
+                        solution_handles.append(self.plots[key][i][0])
+
+                # Add a legend to indicate which plot corresponds to which time
+                ax.legend()
+
             elif variable_lists[0][0].dimensions == 2:
-                # Read dictionary of spatial variables
+                # 2D plot: superimpose plots at different times
                 spatial_vars = self.spatial_variable_dict[key]
-                # there can only be one entry in the variable list
                 variable = variable_lists[0][0]
-                # different order based on whether the domains are x-r, x-z or y-z, etc
-                if self.x_first_and_y_second[key] is False:
-                    x_name = list(spatial_vars.keys())[1][0]
-                    y_name = next(iter(spatial_vars.keys()))[0]
-                    x = self.second_spatial_variable[key]
-                    y = self.first_spatial_variable[key]
-                    var = variable(t_in_seconds, **spatial_vars)
-                else:
-                    x_name = next(iter(spatial_vars.keys()))[0]
-                    y_name = list(spatial_vars.keys())[1][0]
+
+                for _idx, t_single in enumerate(t):
```

Comment:
The index does not appear to be used, so can we just simplify this?
```suggestion
                for t_single in t:
```